src/test/regress/sql/merge_repartition.sql

```diff
@@ -0,0 +1,457 @@
+SHOW server_version \gset
+SELECT substring(:'server_version', '\d+')::int >= 15 AS server_version_ge_15
+\gset
+\if :server_version_ge_15
+\else
+\q
+\endif
+
+-- We create two sets of source and target tables, one set in Postgres and
+-- the other in Citus distributed. We run the _exact_ MERGE SQL on both sets
+-- and compare the final results of the target tables in Postgres and Citus.
+-- The results should match. This process is repeated for various combinations
+-- of MERGE SQL.
+
+DROP SCHEMA IF EXISTS merge_repartition_schema CASCADE;
+CREATE SCHEMA merge_repartition_schema;
+SET search_path TO merge_repartition_schema;
+SET citus.shard_count TO 4;
+SET citus.next_shard_id TO 5000000;
+SET citus.explain_all_tasks TO true;
+SET citus.shard_replication_factor TO 1;
+SET citus.max_adaptive_executor_pool_size TO 1;
+SET client_min_messages = warning;
+SELECT 1 FROM master_add_node('localhost', :master_port, groupid => 0);
+RESET client_min_messages;
+
+
+CREATE TABLE pg_target(id int, val int);
+CREATE TABLE pg_source(id int, val int, const int);
+CREATE TABLE citus_target(id int, val int);
```

Comment:
well, partitioned tables are always interesting. Should we maybe make citus source and target tables partitioned?

Reply:
Done. Added new test `merge_partition_tables.sql`